build-system/amp.extern.js

```diff
@@ -60,6 +60,16 @@ window.AMP_CONFIG.errorReportingUrl;
 
 window.AMP_CONTEXT_DATA;
 
+/**
+ * @typedef {{
+ *   observe: function(!Object),
+ *   disconnect: function(),
+ * }}
+ */
+let PerformanceObserver;
```

Comment:
We currently typically do this a bit differently (our own AMP externs are a different story, since they belong to us).

1. We put them into `third_party/closure-compiler/externs/` folder as a separate file.
2. Link this file from our [compiler config](https://github.com/ampproject/amphtml/blob/8302ba89099dff754f6a028c5fb2940bbcf4982a/build-system/tasks/compile.js#L219).
3. You can use all normal ES6 semantics.

Your extern thus would look something like this:
```
/** See https://w3c.github.io/performance-timeline/# */
class PerformanceObserver {
  /**
    */
  observe(options) {}
  /**
    */
  disconnect() {}  
  ...
  ...
}
```
I think that's all you need. You don't need to add them to `window` - i think it happens automatically by closure.